File: Logic/Python/balanced_parentheses_checker.py
def is_balanced(s):
    stack = []
    match = {')': '(', '}': '{', ']': '['}
    for c in s:
        if c in match.values(): stack.append(c)
        elif not stack or match[c] != stack.pop(): return False
    return not stack

print("Balanced" if is_balanced("{[()]}") else "Not Balanced")

File: Logic/Java/BalancedParenthesesChecker.java
import java.util.*;
public class BalancedParenthesesChecker {
    public static void main(String[] args) {
        String s = "{[()]}";
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if ("({[".indexOf(c) >= 0) stack.push(c);
            else if (!stack.isEmpty() && 
                ((c == ')' && stack.peek() == '(') || 
                 (c == '}' && stack.peek() == '{') || 
                 (c == ']' && stack.peek() == '[')))
                stack.pop();
            else {
                System.out.println("Not Balanced");
                return;
            }
        }
        System.out.println(stack.isEmpty() ? "Balanced" : "Not Balanced");
    }
}